src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializerExt.cs

```diff
@@ -23,17 +24,22 @@ namespace OpenTelemetry.Exporter.Prometheus;
 /// </summary>
 internal static partial class PrometheusSerializer
 {
+    private const int MaxCachedMetrics = 1024;
+
     /* Counter becomes counter
        Gauge becomes gauge
        Histogram becomes histogram
        UpDownCounter becomes gauge
      * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#otlp-metric-points-to-prometheus
     */
-    private static readonly string[] MetricTypes = new string[]
+    private static readonly PrometheusType[] MetricTypes = new PrometheusType[]
     {
-        "untyped", "counter", "gauge", "summary", "histogram", "histogram", "histogram", "histogram", "gauge",
+        PrometheusType.Untyped, PrometheusType.Counter, PrometheusType.Gauge, PrometheusType.Summary, PrometheusType.Histogram, PrometheusType.Histogram, PrometheusType.Histogram, PrometheusType.Histogram, PrometheusType.Gauge,
     };
 
+    private static readonly ConcurrentDictionary<Metric, PrometheusMetric> MetricsCache = new ConcurrentDictionary<Metric, PrometheusMetric>();
```

Comment:
You could use a simple dictionary here. This code would only be run by one thread at a time.

Reply:
How do you know? Couldn't there be multiple requests for the endpoint at the same time?

(I haven't looked closely at how the serializer is used)

Reply:
I see `PrometheusCollectionManager` has a global lock. However, there can be multiple instances of this type. The cache is currently static. It would need to be made non-static to safely use a regular dictionary.

Reply:
> It would need to be made non-static to safely use a regular dictionary.

That should be fine.

Reply:
I've changed it so that the cache is non-static. Because `PrometheusSerializer` itself is static, the cache had to move. It now lives on `PrometheusCollectionManager`. A number of changes then flowed on from that.